R/generate_pt_nowcast_mat_list.R

```diff
@@ -47,33 +49,12 @@ generate_pt_nowcast_mat_list <- function(reporting_triangle_list,
                                          ) - 1,
                                          n = min(
                                            sapply(reporting_triangle_list, nrow)
-                                         )) {
-  if (n > min(
-    sapply(reporting_triangle_list, nrow)
-  )) {
-    cli_abort(
-      message = c(
-        "The number of observations specified for delay estimation is greater ",
-        "than the minimum number of rows in all of the retrospective ",
-        "reporting triangles. Either remove the reporting triangles that do ",
-        "not contain sufficient data, or lower `n_history_delay`"
-      )
-    )
-  }
-  if (n < min(sapply(reporting_triangle_list, ncol))) {
-    cli_abort(
-      message = c(
-        "The number of observations specified for delay estimation is less ",
-        "than one plus the number of columns in the retrospective reporting ",
-        "triangles. The delay distribution can only be estimated from at ",
-        "at least the number of columns in the reporting triangle."
-      )
-    )
-  }
-
+                                         ),
+                                         delay_pmf = NULL) {
   pt_nowcast_mat_list <- lapply(reporting_triangle_list,
     generate_pt_nowcast_mat,
-    n = n
+    n = n,
+    delay_pmf = delay_pmf
```

Comment:
This isn't an argument for the function . If it were to be should it be a list or a single entry. If both should there be tooling to map between these options. max_delay is an argument but isn't passed.

Reply:
Missed `max_delay`. `delay_pmf` was an argument. I have just added support for it to be either a single vector or a list. 

I think we will probably want to add a function that can generate this list of `delay_pmf`s, or at the very least show how one would get that with an `lapply ` on `get_delay_estimate()` iterating across a list of retrospective reporting triangles from presumably a different strata than the list being passed here (which is where the delay pmfs will be applied).